src/common/util/colors.js

```diff
@@ -0,0 +1,19 @@
+import { hexToRgb, rgbToHex } from '@mui/material';
+
+export const interpolateColor = (color1, color2, factor) => {
+  if (factor > 1) factor = 1;
+  if (factor < 0) factor = 0;
+
+  const parseRgb = (rgb) => rgb.match(/\d+/g).map(Number);
```

Comment:
Why do we need this?

Reply:
hexToRgb returns a string, we can change it to this:

const parseRgb = (rgb) => rgb.split(',').map(Number);

Reply:
I removed it, please check